### PR TITLE
feat(chart): chart selection area animation

### PR DESCRIPTION
**Pull request by peter-affenzeller**
### <strong>Pull Request</strong>

Depends on #594 

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ffriedl89** [comment]
Blocked by #594

**Comment by ffriedl89** [comment]
@peter-affenzeller The blocker is now merged - this can be continued :)

**Comment by lukasholzer** [comment]
Please rebase your branch with the latest master, then perform and install and run `yarn format --all` to get the latest formatting changes that were introduced with the version upgrade of prettier to 2.0. This update was necessary to be compatible with the latest typescript version that we are using. Furthermore, we improved the angular template formatting!

**Comment by tomheller** [comment]
@peter-affenzeller you have re-requested the review but the PR is still marked as WIP. 
Do you want us to review?

**Comment by peter-affenzeller** [comment]
@tomheller yes please. it's also no longer blocked since #594 is already merged

File: libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    containerRef: DtChartSelectionAreaContainer,
+  ) {
+    this.overlayRef = overlayRef;
+    this.containerRef = containerRef;
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
```
Comment (lukasholzer) [inline libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts:43]: You don't have Angular lifecycles in pure classes as it is not a part in the Angular Dependency Injection Tree.

You have to handle destroy on your own so please create a destroy method that is called in an `ngOnDestroy` of the wrapping Angular Component.

A class is only used to hold a data state in angular just a better object.

File: libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { OverlayRef } from '@angular/cdk/overlay';
+import { Observable, Subject } from 'rxjs';
+import { DtChartSelectionAreaContainer } from './selection-area-container';
+import { OnDestroy } from '@angular/core';
+
+export class DtChartSelectionAreaOverlayRef implements OnDestroy {
```
Comment (lukasholzer) [inline libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts:22]: Please remove implements as it is a normal class and not a Component or Service – therefore you have No Angular Lifecycle methods.

File: libs/barista-components/chart/src/selection-area/selection-area-container.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  ngOnDestroy(): void {
+    this._destroyed = true;
+  }
+
+  /**
+   * Attach a TemplatePortal as content to this dialog container.
+   * @param portal Portal to be attached as the dialog content.
+   */
+  attachTemplatePortal<C>(portal: TemplatePortal<C>): EmbeddedViewRef<C> {
+    if (this._portalOutlet.hasAttached()) {
```
Comment (heartdisease) [inline libs/barista-components/chart/src/selection-area/selection-area-container.ts:69]: From what I've heard, the new policy is to never break a page by throwing errors. Maybe we should also consider logging an error message here instead?

**Review by heartdisease — CHANGES_REQUESTED** [review]

**Review by rowa-audil — APPROVED** [review]

**Comment by peter-affenzeller** [comment]
due to the work on the new components we decided to not continue working on this one.

**Pull request closed without merging**